server/src/main/java/org/opensearch/index/translog/Translog.java

```diff
@@ -624,7 +629,7 @@ public Snapshot newSnapshot() throws IOException {
      * @param toSeqNo   the upper bound of the range (inclusive)
      * @return the new snapshot
      */
-    public Snapshot newSnapshot(long fromSeqNo, long toSeqNo) throws IOException {
+    public Snapshot newSnapshot(long fromSeqNo, long toSeqNo, boolean requiredFullRange) throws IOException {
```

Comment:
When won't you get the full range from tlog?

Reply:
This is to keep on parity with Lucene based fetch and ensure we are able to get all the operations in the requested range.